File: 0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for (char p : s.toCharArray()) {
            if (p == '(' || p == '[' || p == '{') {
                stack.push(p);
            } else {
                if (stack.isEmpty()) return false;
                char top = stack.pop();
                if ((p == ')' && top != '(') || 
                    (p == ']' && top != '[') || 
                    (p == '}' && top != '{')) {
                    return false;
                }
            }
        }
        return stack.isEmpty();
    }
}

File: 0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for (auto p:s){
            if (p=='(' || p=='{' || p=='['){
                st.push(p);
            }else{
                if(st.empty()){
                    return false;
                    break;
                }
                char top = st.top();
                st.pop();

                if ( p == ')' && top != '(' || p == '}' && top != '{' || p == ']' && top != '['){
                    return false;
                }
            }
        }
        return (st.empty());
    }
};

